File: src/main/java/site/gachontable/gachontablebe/domain/waiting/usecase/CreateWaiting.java
package site.gachontable.gachontablebe.domain.waiting.usecase;

import site.gachontable.gachontablebe.domain.auth.domain.AuthDetails;
import site.gachontable.gachontablebe.domain.waiting.presentation.dto.request.RemoteWaitingRequest;
import site.gachontable.gachontablebe.domain.waiting.presentation.dto.response.WaitingResponse;

public interface CreateWaiting {
    WaitingResponse execute(AuthDetails authDetails, RemoteWaitingRequest request, String lockKey);

//    WaitingResponse execute(AuthDetails authDetails, OnsiteWaitingRequest request, String lockKey);
}

File: src/main/java/site/gachontable/domain/waiting/port/in/CreateWaiting.kt
package site.gachontable.domain.waiting.port.in;

import site.gachontable.infra.security.principal.AuthDetails;
import site.gachontable.presentation.waiting.dto.request.RemoteWaitingRequest;
import site.gachontable.presentation.waiting.dto.response.WaitingResponse;

public interface CreateWaiting {

    WaitingResponse execute(AuthDetails authDetails, RemoteWaitingRequest request, String lockKey);
}
